atomic_reactor/plugins/pre_resolve_composes.py

```diff
@@ -258,19 +267,16 @@ def resolve_signing_intent(self):
         self.compose_config.set_signing_intent(signing_intent['name'])
 
     def forward_composes(self):
-        repos_by_arch = {}
+        repos_by_arch = defaultdict(list)
         # set overrides by arch if arches are available
         for compose_info in self.composes_info:
             for arch in compose_info.get('arches') or []:
-                repos_by_arch.setdefault(arch, [])
                 repos_by_arch[arch].append(compose_info['result_repofile'])
+            else:
+                repos_by_arch[None].append(compose_info['result_repofile'])
```

Comment:
There's no `break` in the loop, so this `else:` block will *always* run, which doesn't look like what you mean.

Reply:
Fixed. Now pulling arches for pulp from the pulp entries.